provider/labels.go

```diff
@@ -31,17 +31,16 @@ func fixEmptyLabels(_ context.Context, req shimv2.PlanStateEditRequest) (cty.Val
 
 	// Apply default labels first.
 	if pConfig := resource.FromResourcePropertyValue(resource.NewProperty(req.ProviderConfig)); pConfig.IsMap() {
-		l := pConfig.AsMap()["defaultLabels"]
-		if l.IsMap() {
-			programLabels = l.AsMap()
+		if labels, ok := pConfig.AsMap().GetOk("defaultLabels"); ok && labels.IsMap() {
+			programLabels = labels.AsMap()
 		}
 	}
 
 	// Apply labels next, allowing labels to override defaultLabels.
 	if inputs, ok := (resource.PropertyPath{labelsPropertyName}.Get(resource.NewProperty(req.NewInputs))); ok {
 		if labels := resource.FromResourcePropertyValue(inputs); labels.IsMap() {
-			for k, v := range labels.AsMap() {
-				programLabels[k] = v
+			for k, v := range labels.AsMap().AsMap() {
```

Comment:
This looks suspect, is there some context on what is changing for label handling? Thank you.

Reply:
The behavior hasn't changed but the public API of `property.Value` has. You could no longer iterate over `property.Map` directly (the result of `labels.AsMap()`) but you need to call `.AsMap()` again so that the expression becomes `map[string]property.Value` which you can iterate on

Reply:
yeah! so @Zaid-Ajaj may have a bit more context here, but I believe it's related to the change in https://github.com/pulumi/pulumi/pull/17500.

The first `AsMap` returns a `Map` from Value but as of the above change to pu/pu, we can no longer range over a `Map`. We must retrieve the underlying map (also called `AsMap()`) in order to retrieve a map[string]Value that we can range over.